collector/monitors/failed_redelegations_monitor.go

```diff
@@ -78,36 +79,58 @@ func (m *FailedRedelegationsMonitor) Handler(ctx context.Context) error {
 		return fmt.Errorf("failed to get whiltelisted whitelistedValidators for %s: %w", m.Name(), err)
 	}
 
-	delegationsResponse, err := m.apiClient.Staking.GetStakingDelegatorsDelegatorAddrDelegations(&staking.GetStakingDelegatorsDelegatorAddrDelegationsParams{
-		DelegatorAddr: m.hubAddress,
-		Context:       ctx,
-	})
-	if err != nil {
-		return fmt.Errorf("failed to get whitelistedValidators of delegator: %w", err)
-	}
+	var paginationKey strfmt.Base64
+	for {
+		delegationsResponse, err := m.apiClient.Query.DelegatorDelegations(&query.DelegatorDelegationsParams{
+			PaginationKey: &paginationKey,
+			DelegatorAddr: m.hubAddress,
+			Context:       ctx,
+		})
+		if err != nil {
+			return fmt.Errorf("failed to get whitelistedValidators of delegator: %w", err)
+		}
 
-	if err := delegationsResponse.GetPayload().Validate(nil); err != nil {
-		return fmt.Errorf("failed to validate delegator's validators response: %w", err)
-	}
+		if err := delegationsResponse.GetPayload().Validate(nil); err != nil {
+			return fmt.Errorf("failed to validate delegator's validators response: %w", err)
+		}
+
+		if delegationsResponse.Payload == nil {
+			return fmt.Errorf("failed to validate delegator's validators response: %w", err)
+		}
 
-	for _, delegation := range delegationsResponse.GetPayload().Result {
-		if err := delegation.Validate(nil); err != nil {
-			return fmt.Errorf("failed to validate delegation: %w", err)
+		if delegationsResponse.Payload.Pagination != nil {
+			paginationKey = delegationsResponse.Payload.Pagination.NextKey
 		}
 
-		delegatedAmount := uint64(0)
-		if delegation.Balance != nil {
-			delegatedAmount, err = strconv.ParseUint(delegation.Balance.Amount, 10, 64)
-			if err != nil {
-				return fmt.Errorf("failed to parse delegation amount: %w", err)
+		for _, response := range delegationsResponse.GetPayload().DelegationResponses {
+			if err := response.Validate(nil); err != nil {
+				return fmt.Errorf("failed to validate response: %w", err)
+			}
+
+			if response.Delegation == nil {
+				return fmt.Errorf("failed to validate response: delegaion is nil")
+			}
+
+			delegatedAmount, ok := types.NewInt(0), false
+			if response.Balance != nil {
+				if delegatedAmount, ok = types.NewIntFromString(response.Balance.Amount); !ok {
+					return fmt.Errorf("failed to parse delegation balance amount: %w", err)
+				}
+			} else {
+				return fmt.Errorf("failed to get response balance: balance is nil")
+			}
+
+			tmpMetricVectors[FailedRedelegations].Set(response.Delegation.ValidatorAddress, 0)
+
+			// if delegated amount is greater than zero and the whitelisted validators don't contain a validator
+			// that means a redelegation was not successful
+			if !delegatedAmount.IsZero() && !contains(whitelistedValidators, response.Delegation.ValidatorAddress) {
+				tmpMetricVectors[FailedRedelegations].Set(response.Delegation.ValidatorAddress, 1)
```

Comment:
isnt better to use monikers?

Reply:
The task says _New monitor is implemented using labels to show specific validator addresses on the graph_, and imo addresses are better than monikers for this panel particularly. But let's clarify this with @turinkay and @oopcode

Reply:
if it's possible and not too costly, we should add a moniker in parenthesis: address (moniker)